bot/exts/moderation/duty.py

```diff
@@ -0,0 +1,135 @@
+import datetime
+import logging
+
+from async_rediscache import RedisCache
+from dateutil.parser import isoparse
+from discord import Member
+from discord.ext.commands import Cog, Context, group, has_any_role
+
+from bot.bot import Bot
+from bot.constants import Emojis, Guild, MODERATION_ROLES, Roles
+from bot.converters import Expiry
+from bot.utils.scheduling import Scheduler
+
+
+log = logging.getLogger(__name__)
+
+
+class Duty(Cog):
+    """Commands for a moderator to go on and off duty."""
+
+    # RedisCache[str, str]
+    # The cache's keys are mods who are off-duty.
+    # The cache's values are the times when the role should be re-applied to them, stored in ISO format.
```

Comment:
I remember Mark said somewhere else that using Unix timestamp is better.

Reply:
Is there a specific reason? in terms of functionality it seems to work just fine.

Reply:
[A previous discussion on that is here](https://github.com/python-discord/bot/pull/987#discussion_r436229629), although I feel it's fine either way.

Reply:
Yeah, I would prefer timestamps over formatted strings (now for consistency too since it's a more established pattern) but it's not a dealbreaker.